### DISCUSSION TITLE
[Thanos] Compaction/Downsampling: build chunks from object storage without touching disks

### DISCUSSION THREAD

**Discussion by nate-double-u** [cncf/mentoring#614] [Google Summer of Code 2022]
Right now during the compaction/downsampling stage, the Thanos Compactor always downloads TSDB blocks to local disk first and compacts/downsamples them later. This download process takes a long time if the data is large. Actually it is doable to read the data we need from the object storage directly and perform the action on the fly. This improvement will help save a lot of disk space in larger deployment.

Mentor(s): Ben Ye (@yeya24), Matej Gera (@matej-g)

https://github.com/cncf/mentoring/blob/main/summerofcode/2022.md#compactiondownsampling-build-chunks-from-object-storage-without-touching-disks

**Comment by rvacaru** [comment]
Hi Thanos people,
I'm Razvan Vacaru, a data engineer based in Amsterdam. I'm looking into the GSoC 2022 ideas and this is one that I think I'll like and I'll be able to help with.

In short my CV is:

- Bachelor's degree in CS Engineering (Java, C, Javascript, Linux sysadmin)
- Worked ~6 years mostly as a Java/Spring developer (4 of the 6 years in AWS)
- Working for ~ 2 years as a Data engineer with python, spark, terraform, datadog, ...
- Knowledge of distributed systems like cassandra, spark and hadoop
- Check the full CV in LinkedIn: https://www.linkedin.com/in/razvan-vacaru-17787b5a/
- From time to time I also do small contributions to OS projects like akka or dbt: https://github.com/rvacaru

My goal is to increase my knowledge of the K8s ecosystem, distributed systems, monitoring and Golang.
Do you think I would be a good fit for this project?
